### ISSUE TITLE
Error: EPERM: operation not permitted, copyfile './db/kuma.db' -> './data/kuma.db'

### ISSUE THREAD

**Issue by D3Brooksster** [louislam/uptime-kuma#3291]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

I'm seeing this issue on first run.

Not sure where the issue is coming from; I've checked the volume exists and has sufficient space. Sorry if I'm missing something obvious here but can't see any refence to similar issues. 

Unable to troubleshoot further due to container crashing following error.

YML:
version: '3.3'

services:
  uptime-kuma:
    image: louislam/uptime-kuma:latest
    container_name: uptime-kuma_latest
    volumes:
      - /share/Containers/container-station-data/lib/docker/volumes/uptimekuma_config/_data:/app/data
    ports:
      - 3001:3001
    restart: unless-stopped
networks:
  default:
   external:
     name: Docker Bridge

### 👟 Reproduction steps

Start container stack

### 👀 Expected behavior

Stack starts as expected and error is not shown

### 😓 Actual Behavior

Container crashes on start-up

### 🐻 Uptime-Kuma Version

1.21.3

### 💻 Operating System and Arch

Docker

### 🌐 Browser

n/a

### 🐋 Docker Version

Docker 17.09.1-ce (QNAP Container Station)

### 📝 Relevant log output

```shell
2023-06-20T18:50:47.178648928Z 2023-06-20T19:50:47+01:00 [SERVER] INFO: Version: 1.21.3
2023-06-20T18:50:47.243450654Z 2023-06-20T19:50:47+01:00 [DB] INFO: Data Dir: ./data/
2023-06-20T18:50:47.244016234Z 2023-06-20T19:50:47+01:00 [SERVER] INFO: Copying Database
2023-06-20T18:50:47.248693183Z Trace: Error: EPERM: operation not permitted, copyfile './db/kuma.db' -> './data/kuma.db'
2023-06-20T18:50:47.248717519Z     at Object.copyFileSync (node:fs:2847:3)
2023-06-20T18:50:47.248721405Z     at initDatabase (/app/server/server.js:1658:12)
2023-06-20T18:50:47.248732955Z     at /app/server/server.js:173:11
2023-06-20T18:50:47.248736633Z     at Object.<anonymous> (/app/server/server.js:1565:3)
2023-06-20T18:50:47.248740233Z     at Module._compile (node:internal/modules/cjs/loader:1196:14)
2023-06-20T18:50:47.248743515Z     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
2023-06-20T18:50:47.248746670Z     at Module.load (node:internal/modules/cjs/loader:1074:32)
2023-06-20T18:50:47.248749809Z     at Function.Module._load (node:internal/modules/cjs/loader:909:12)
2023-06-20T18:50:47.248756031Z     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
2023-06-20T18:50:47.248759491Z     at node:internal/main/run_main_module:22:47 {
2023-06-20T18:50:47.248762607Z   errno: -1,
2023-06-20T18:50:47.248765626Z   syscall: 'copyfile',
2023-06-20T18:50:47.248768646Z   code: 'EPERM',
2023-06-20T18:50:47.248771611Z   path: './db/kuma.db',
2023-06-20T18:50:47.248774583Z   dest: './data/kuma.db'
2023-06-20T18:50:47.248777516Z }
2023-06-20T18:50:47.248780423Z     at process.<anonymous> (/app/server/server.js:1804:13)
2023-06-20T18:50:47.248783625Z     at process.emit (node:events:513:28)
2023-06-20T18:50:47.248786650Z     at emit (node:internal/process/promises:140:20)
2023-06-20T18:50:47.248789693Z     at processPromiseRejections (node:internal/process/promises:274:27)
2023-06-20T18:50:47.248792815Z     at processTicksAndRejections (node:internal/process/task_queues:97:32)
2023-06-20T18:50:47.249522848Z If you keep encountering errors, please report to https://github.com/louislam/uptime-kuma/issues
```

**Comment by CommanderStorm** [comment]
So basically you have `/app/data` mounted externally which leads to permission problems right?
What are the file+folder permissions in `/share/Containers/container-station-data/lib/docker/volumes/uptimekuma_config/_data` ?

Are they writable by the `root` user?

Also I would not recommend pinning to latest as you did in your deployment.
Two reasons:
- you do not know which version you are running exactly
- If we decide to do a Major version bump (e.g. the upcoming v2 release) you will get this upon the next restart after the release without the ability to run backups and to monitor for defects.

**Comment by rezzorix** [comment]
[Shared volumes do not work. Use local volumes only.](https://github.com/louislam/uptime-kuma#-docker)

https://github.com/louislam/uptime-kuma#-docker

**Comment by D3Brooksster** [comment]
Thank you both; I did see the point regarding shared volumes but as this is a local volume to the Docker instance I didn't know this would still apply.

Let me try again removing the full path as is in the docker compose example.

In terms of permissions this path is owned and writable by the root user and is how all of my other containers are configured.

**Comment by D3Brooksster** [comment]
Hi,

I tried again based on your feedback and have the following docker compose file -using the volume mapping as described [here](https://github.com/louislam/uptime-kuma/blob/master/docker/docker-compose.yml).

I've also updated this to use version 1 rather then latest. 

version: '3.3'

services:
  uptime-kuma:
    image: louislam/uptime-kuma:1
    container_name: uptime-kuma_1
    volumes:
      - ./uptime-kuma-data:/app/data
    ports:
      - 3001:3001
    restart: unless-stopped
networks:
  default:
   external:
     name: Docker Bridge

I'm still getting the same error as above but can now see the volume has been mapped to - /Containers/container-station-data/application/uptimekuma/uptime-kuma-data

[/share/Containers/container-station-data/application/uptimekuma] # ls -ltr
drwxr-xr-x 3 admin administrators 4096 2023-06-21 20:20 uptime-kuma-data/
# id -a
uid=0(admin) gid=0(administrators) groups=0(administrators),100(everyone)

I also tried adding the following environment options to the compose file to ensure this was running with the correct PID

    environment:
      - PUID=0
      - PGID=0

Any help would be great apricated and thanks for the advice this far.

**Comment by D3Brooksster** [comment]
Just as a folllow-up to this I also tried just running this directly using the following but still see the same error:

docker run -d --restart=always -p 3001:3001 -v uptime-kuma:/app/data --name uptime-kuma louislam/uptime-kuma:1

**Comment by CommanderStorm** [comment]
> I also tried just running this directly using the following but still see the same error

The problem with that is, that it is [not reproducible](https://stackoverflow.com/help/minimal-reproducible-example) on my machine.
What kind of system are you running on:
- OS ([is it a mac](https://stackoverflow.com/a/72496112)? Or RHEL?)
- `docker --version`
- Seccomp policies

**Comment by D3Brooksster** [comment]
Sure;

Running from container station from QNAP:

[~] #   cat /etc/os-release
NAME="QTS"
VERSION="4.3.6 (20221124)"
ID=qts
PRETTY_NAME="QTS 4.3.6 (20221124)"
VERSION_ID="4.3.6"

Docker version controlled via QNAP application and unable to be updated (at least not that I'm aware of)

[~] # docker --version
Docker version 17.09.1-ce, build 0bbe3ac

No SECCOMP policies applied based on what I can see here;

sudo grep SECCOMP /boot/config-$(uname -r)
grep: /boot/config-4.2.8: No such file or directory
[~] # grep CONFIG_SECCOMP= /boot/config-$(uname -r)
grep: /boot/config-4.2.8: No such file or directory

**Comment by rezzorix** [comment]
What QNAP do you have exactly?

**Comment by CommanderStorm** [comment]
This seems to be sort of common, at least I found this thread (in German, only result on Google which is relevant)
https://forum.qnapclub.de/thread/47439-operation-not-permitted-wenn-nicht-root-nutzer/

**Comment by CommanderStorm** [comment]
This seems to be sort of common, at least I found this thread (in German, only result on Google which is relevant)
https://forum.qnapclub.de/thread/47439-operation-not-permitted-wenn-nicht-root-nutzer/
